File: pivot.pl
#!/usr/bin/perl

use 5.016;
use DDP;
use integer;
my @a = @{[0,1]};
p @a;

sub return_pivod(@) {
	my @arr = @_;
	return -1 unless (@arr);
	
	my $pivot = 0;
	my $sum_left = 0;
	my $sum_right = 0;
	for (1..$#arr) {
		$sum_right += $arr[$_];
	}
	while ($sum_left != $sum_right) {
		last if $pivot + 1 > $#arr;
		$sum_left += $arr[$pivot];
		$sum_right -= $arr[$pivot + 1];
		$pivot++;
		
	}
	return $pivot > $#arr ? -1 : $pivot;
}

say return_pivod(@a);
reverse

File: pivot.py
#!/usr/bin/python3
nums = [0,1]
class Solution:
    def pivotIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if (len(nums) == 0):
            return -1
        pivot = 0
        sum_left = 0
        sum_right = sum(nums[1:])
        while(sum_left != sum_right):
            if (pivot + 1 > len(nums) - 1):
                break
            sum_left += nums[pivot]
            sum_right -= nums[pivot + 1]
            pivot = pivot + 1

        if (sum_right != sum_left):
            pivot = -1
        return pivot 

p = Solution()
print (p.pivotIndex(nums))
